ios/RNSScreen.m

```diff
@@ -612,6 +636,53 @@ - (void)notifyFinishTransitioning
   [RNSScreenWindowTraits updateWindowTraits];
 }
 
+# pragma mark - transition progress related methods
+
+- (void)setupProgressNotification
+{
+  if (self.transitionCoordinator != nil && !_isSendingProgress) {
+    _isSendingProgress = YES;
+    UIView *fakeView = [UIView new];
+    fakeView.alpha = 0.0;
+    _fakeView = fakeView;
+    [self.transitionCoordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+      [[context containerView] addSubview:fakeView];
+      fakeView.alpha = 1.0;
+      self->_animationTimer = [CADisplayLink displayLinkWithTarget:self selector:@selector(handleAnimation)];
+      [self->_animationTimer addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSDefaultRunLoopMode];
+    } completion:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+      [self->_animationTimer setPaused:YES];
+      [self->_animationTimer invalidate];
+      self->_fakeView = nil;
+      [fakeView removeFromSuperview];
+      self->_isSendingProgress = NO;
+    }];
+  }
+}
+
+- (void)handleAnimation
+{
+  if ([[_fakeView layer] presentationLayer] != nil) {
+    CGFloat fakeViewAlpha = _fakeView.layer.presentationLayer.opacity;
+    if (_currentAlpha != fakeViewAlpha) {
+      _currentAlpha = fmax(0.0, fmin(1.0, fakeViewAlpha));
+      [self notifyTransitionProgress:_currentAlpha closing:_closing];
+    }
+  }
+}
+
+- (void)notifyTransitionProgress:(double)progress closing:(BOOL)closing
+{
+  [((RNSScreenView *)self.view) notifyTransitionProgress:progress closing:closing];
+  // if we are in a modal, we want to send transition to the above screen too, which might not trigger appear/disappear events
+  // e.g. when we are in iOS >= 13 default modal
+  // we also check if we are not already sending the progress of transition and if it does not present another modal,
+  // because otherwise we would send progress to the screen 2 levels higher then the current one
```

Comment:
I don't understand this. Why would this method send events to the view that is not transitioning at the moment?

Reply:
The explanation is the same as in https://github.com/software-mansion/react-native-screens/pull/890#discussion_r646561799.